src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1545,11 +1545,17 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
     uct_dc_mlx5_update_tx_res(iface, txwq, txqp, pi);
     uct_rc_txqp_purge_outstanding(&iface->super.super, txqp, ep_status, pi, 0);
 
+    /* Invoke a user's error callback and release TX/FC resources before
+     * releasing DCI, to have DCI for doing possible flush(CANCEL) */
```

Comment:
maybe move the assert in line 1550 into uct_dc_mlx5_iface_dci_put?

Reply:
done